docs/admin/env_variables.md

```diff
@@ -45,6 +45,7 @@ Environment variables in this section only apply to the server side.
 |DAOS\_SCHED\_PRIO\_DISABLED|Disable server ULT prioritizing. BOOL. Default to 0.|
 |DAOS\_SCHED\_RELAX\_MODE|The mode of CPU relaxing on idle. "disabled":disable relaxing; "net":wait on network request for INTVL; "sleep":sleep for INTVL. STRING. Default to "net"|
 |DAOS\_SCHED\_RELAX\_INTVL|CPU relax interval in milliseconds. INTEGER. Default to 1 ms.|
+|DAOS\_STRICT\_SHUTDOWN|Use the strict mode when shutting down engines. BOOL. Default to 0. In the strict mode, when certain resource leaks are detected, for instance, the engine will raise an assertion failure.|
```

Comment:
Do we need this?  If you're only setting it for NLT and that will fail the test on new D_ERROR calls anyway then I'm not sure the additional complexity is worth it.

Reply:
@ashleypittman, the idea is to set it for more, non-NLT tests going forward, to prevent regressions.

Reply:
Ok, then it's worth keeping if you've a use in mind.  Do we need it on for NLT, the test should fail either way and the assert would make it clearer what the issue was but it might also mask other failures compared to logging and an orderly shutdown.

Reply:
@ashleypittman, so what do you think we should do (or not do) with regard to NLT?

Reply:
I think either would be fine.  I'm not exactly sure what sort of regression you're looking for.  Given this bug has existed essentially forever the fact that this PR to fix it is so small indicates that this isn't something that's going to regress frequently.

That said, NLT is only a small part of testing and ftest doesn't scan logs for errors it would be interesting to set this variable for ftest and see if we're still experiencing issues.  Without having done that we won't know if we can close the ticket or not.

Reply:
The regressions I have in mind are, for instance, someone adds a new function that looks up a ds_pool_hdl (or a ds_pool/ds_pool_child) but forgets to put the reference. These sound likely to me, especially if we consider that other objects that hold ds_pool/ds_pool_child references could be leaked themselves. The pool_tls_fini check in "strict" mode will make such regressions more evident. Since it's impractical to manually check all possibilities like those regularly, we might want to try gradually "seize" more and more tests (when CI testing becomes stable and quick). If we don't protect the efforts in fixing these issues, I'm afraid what we're doing here might not meaningful...

If there's better and practical protection method, I'd be happy to see us switch to it.

Reply:
Ok, so the plan is to enable this flag gradually to extend test coverage, fix anything that we discover broken and leave it enabled to prevent regressions, but to not do so until CI is more stable?

Reply:
Yes.